### PR TITLE
Type checking to avoid fatal error

### PR DESCRIPTION
**Pull request by morille**
Some of my users had fatal errors while using the API because they made mistake in the Media Form.
This check allow to return a clear description of the problem.

### CODE REVIEW

**Comment by OskarStark** [comment]
looks good to me, but can you please check travis? thank you

**Comment by chlabelek** [comment]
I fixed the PHP compatibility issues.
Now there are 2 remaining errors but they do not depend on me I think.

**Comment by morille** [comment]
Rebased.

File: Provider/FileProvider.php [hunk 1]
```diff
@@ -192,6 +192,15 @@ protected function fixBinaryContent(MediaInterface $media)
             return;
         }
 
+        // is_file will cause a fatal error if binary content is not a string
+        if (!is_string($media->getBinaryContent())) {
```
Comment (OskarStark) [inline Provider/FileProvider.php]: what if `$media->getBinaryContent() == Symfony\Component\HttpFoundation\File\File`
does `is_string()` return `true`?

ping @greg0ire

Comment (greg0ire) [inline Provider/FileProvider.php]: Do you literally mean the class name or an object of this class?

Comment (OskarStark) [inline Provider/FileProvider.php]: I mean, what if this returns an object?

Comment (greg0ire) [inline Provider/FileProvider.php]: I don't think so.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**